src/main/java/graphql/Profiler.java

```diff
@@ -0,0 +1,57 @@
+package graphql;
+
+import graphql.execution.EngineRunningObserver;
+import graphql.execution.ResultPath;
+import graphql.language.OperationDefinition;
+import graphql.schema.DataFetcher;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@Internal
+@NullMarked
+public interface Profiler {
```

Comment:
Add a quick JavaDoc explaining what this does

> A Profiler monitors important execution timings and counters. Enable this via ExecutionInput/new Unusual Configuration method

Of course 1 sentence isn't enough, let's add a documentation link here when it's written

Reply:
This is an internal class btw